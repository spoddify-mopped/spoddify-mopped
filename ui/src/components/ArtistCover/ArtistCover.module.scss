.cover {
  img {
    border-radius: 180px;
  }

  span:last-child {
    margin-bottom: 1rem !important;
  }
}
