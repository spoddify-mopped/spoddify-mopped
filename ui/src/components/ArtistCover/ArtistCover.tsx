import { Artist } from '../../clients/api.types';
import Cover from '../Cover/Cover';
import React from 'react';
import styles from './ArtistCover.module.scss';

type Props = {
  item: Artist;
  onClick: (id: string) => void;
};

const ArtistCover = ({ item, onClick }: Props): React.ReactElement => {
  return (
    <Cover
      className={styles.cover}
      item={{
        id: item.id,
        image: item.imageUrl,
        name: item.name,
        onClick: () => {
          onClick(item.id);
        },
      }}
    />
  );
};

export default ArtistCover;
