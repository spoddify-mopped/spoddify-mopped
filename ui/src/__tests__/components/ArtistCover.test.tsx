import ArtistCover from '../../components/ArtistCover/ArtistCover';
import React from 'react';
import renderer from 'react-test-renderer';

it('ArtistCover renders with required properties', () => {
  const tree = renderer
    .create(
      <ArtistCover
        item={{
          id: 'some id',
          name: 'some name',
        }}
        onClick={() => ''}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('ArtistCover renders with image', () => {
  const tree = renderer
    .create(
      <ArtistCover
        item={{
          id: 'some id',
          imageUrl: 'some image url',
          name: 'some name',
        }}
        onClick={() => ''}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
