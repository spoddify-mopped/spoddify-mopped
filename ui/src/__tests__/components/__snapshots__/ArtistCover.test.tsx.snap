// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistCover renders with image 1`] = `
<div
  className="cover cover"
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="some image url"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  />
</div>
`;

exports[`ArtistCover renders with required properties 1`] = `
<div
  className="cover cover"
  onClick={[Function]}
>
  <img
    alt="Thumb"
    src="cover_replacement.png"
  />
  <span
    className="title"
    title="some name"
  >
    some name
  </span>
  <span
    className="subtitle"
  />
</div>
`;
